src/CartesianProduct.jl

```diff
@@ -562,3 +563,147 @@ function same_block_structure(x::AbstractVector{S1}, y::AbstractVector{S2}
+index of the `i`-th block, depending on whether one of the block's dimension indices is specified in `vars`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+- `vars` -- list containing the variables of interest, sorted in ascending order
+
+### Output
+
+A vector of tuples, where values in tuple relate to range of dimensions in the i-th block.
+
+### Example
+
+```jldoctest
+julia> using LazySets: block_to_dimension_indices
+
+julia> cpa = CartesianProductArray([BallInf(zeros(n), 1.0) for n in [1, 3, 2, 3]]);
+
+julia> block_to_dimension_indices(cpa, [2, 4, 8])
+(Tuple{Int64,Int64}[(-1, -1), (2, 4), (-1, -1), (7, 9)], 2)
+```
+This vector represents the mapping "second block from dimension 2 to dimension 4,
+fourth block from dimension 7 to dimension 9."
+These blocks contain the dimensions specified in `[2, 4, 8]`.
+"""
+function block_to_dimension_indices(cpa::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+    result = fill((-1, -1), length(array(cpa)))
+    non_empty_length = 0
+    start_index, end_index = 1, 0
+    v_i = 1
+    for i in 1:length(cpa.array)
+        end_index += dim(cpa.array[i])
+        while v_i <= length(vars) && vars[v_i] <= end_index
+            if vars[v_i] ∈ (start_index : end_index)
+                result[i] = (start_index, end_index)
+                non_empty_length += 1
+                while v_i <= length(vars) && vars[v_i] <= end_index
+                    v_i +=1
+                end
+                break
+            end
+        end
+        if v_i > length(vars)
+            break
+        end
+        start_index = end_index + 1
+    end
+    return result, non_empty_length
+end
+
+#same but for all variables
+function block_to_dimension_indices(cpa::CartesianProductArray{N}) where {N}
+    result = Vector{Tuple{Int, Int}}(undef, length(cpa.array))
+
+    start_index, end_index = 1, 0
+    for i in 1:length(cpa.array)
+        end_index += dim(cpa.array[i])
+        result[i] = (start_index, end_index)
+        start_index = end_index + 1
+    end
+    return result, length(cpa.array)
+end
+
+"""
+    substitute_blocks(low_dim_cpa::CartesianProductArray{N},
+                         orig_cpa::CartesianProductArray{N},
+                           blocks::Vector{Tuple{Int,Int}}) where {N}
+
+Return merged Cartesian Product Array between original CPA and some low-dimensional CPA,
+which represents updated subset of variables in specified blocks.
+
+### Input
+
+- `low_dim_cpa` -- low-dimensional cartesian product array
+- `orig_cpa` -- original high-dimensional Cartesian product array
+- `blocks` -- index of the first variable in each block of `orig_cpa`
+
+### Output
+
+Merged cartesian product array
+
+"""
+function substitute_blocks(low_dim_cpa::CartesianProductArray{N},
+                              orig_cpa::CartesianProductArray{N},
+                              blocks::Vector{Tuple{Int,Int}}) where {N}
```

Comment:
indentation (also in the comment header above)

Reply:
I believe for the comment header above it is good. Should it be aligned to the semicolon?

Reply:
It should be aligned with the `(`.